### PR TITLE
[expo-file-system][docs] Add `uploadAsync` to docs file

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

I've forgotten to update docs when I've finished this task: https://github.com/expo/expo/issues/5841.

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/filesystem.md [hunk 1]
```diff
@@ -115,6 +115,48 @@ These values can be used to define how data is read / written.
...
+- **FileSystem.FileSystemSessionType.BACKGROUND** -- Using this mode means that the downloading/uploading session on the native side will work even if the application is moved to background. If the task completes while the application is in background, the Promise will be either resolved immediately or (if the application execution has already been stopped) once the app is moved to foreground again.
+- **FileSystem.FileSystemSessionType.FOREGROUND** -- Using this mode means that downloading/uploading session on the native side will be terminated once the application becomes inactive (e.g. when it goes to background). Bringing the application to foreground again would trigger Promise rejection.
+
+### `FileSystem.FileSystemUploadOptions`
+
+- **FileSystem.FileSystemUploadOptions.BINARY_CONTENT** -- The file will be sent as a request's body. The request can't contain additional data.
+
+- **FileSystem.FileSystemUploadOptions.MULTIPART** -- An [RFC 2387-compliant](https://www.ietf.org/rfc/rfc2387.txt) request body. The provided file will be encoded into HTTP request. This request can contain additional data.
+
+#### How handel those request?
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/filesystem.md:131]:
```suggestion
#### How to handle such requests?
```

File: docs/pages/versions/unversioned/sdk/filesystem.md [hunk 2]
```diff
@@ -115,6 +115,48 @@ These values can be used to define how data is read / written.
...
+
+The simple server in `node.js`, which can save uploaded images on disk:
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/filesystem.md:133]:
```suggestion
The simple server in Node.js, which can save uploaded images to disk:
```

File: docs/pages/versions/unversioned/sdk/filesystem.md [hunk 3]
```diff
@@ -284,6 +328,44 @@ Returns a Promise that resolves to an object with the following fields:
...
+
+  - **uploadType (_FileSystemUploadOptions_)** -- Upload type determines how the file will be sent to the server. Default to `FileSystemUploadType.BINARY_CONTENT`.
+
+  If `uploadType` is equal `FileSystemUploadType.MULTIPART`, more options are available:
+
+  - **fieldName (_string_)** -- The name of the field which will hold uploaded file. Defaults to the file name without an extension.
+
+  - **mimeType (_string_)** -- The MIME type of the provided file. If not provided, the module will try to guess it based on the extension.
+
+  - **parameters (_Record<string, string>_)** -- Additional HTTP parameters.
```
Comment (sjchmiela) [inline docs/pages/versions/unversioned/sdk/filesystem.md:357]: By additional HTTP parameters you mean other form properties? Like… for `fieldName: "image"` under `"image"` there would be an image and here I could pass `{ name: "Stanley" }` to have `"Stanley"` under `"name"`?

Comment (lukmccall) [reply docs/pages/versions/unversioned/sdk/filesystem.md]: Exactly ;)

**Review by tsapeta — APPROVED** [review]

**Review by sjchmiela — APPROVED** [review]
👌

**Pull request merged**